src/pydocstyle/parser.py

```diff
@@ -576,12 +576,17 @@ def parse_definition(self, class_):
     def parse_skip_comment(self):
         """Parse a definition comment for noqa skips."""
         skipped_error_codes = ''
-        if self.current.kind == tk.COMMENT:
-            if 'noqa: ' in self.current.value:
-                skipped_error_codes = ''.join(
-                     self.current.value.split('noqa: ')[1:])
-            elif self.current.value.startswith('# noqa'):
-                skipped_error_codes = 'all'
+        while self.current.kind in (tk.COMMENT, tk.NEWLINE, tk.NL):
```

Comment:
I've tried all kinds of empty files, files with only comments etc. Nothing unexpected occurs. My understanding is that if the file has only comments, this code won't be called anyway.

As for adding a test for a file with all comments, I don't see that fitting in with the current testing setup. This is because the test case files are expected to have an `expectation` variable defined. Can't do that in an empty file. If you want such a test, how would you go about implementing it?

Reply:
You can add an integration test https://github.com/PyCQA/pydocstyle/blob/master/src/tests/test_integration.py#L333

See above for example. You can add tests with an empty file and check that the output code is 0

Reply:
I added two integration tests:

1. Test that a file with only comments does only cause a D100.
2. Test that a file with comments and docstring does not throw an error.